deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +387,185 @@ public boolean cleanUp() {
 	public int getWeight() {
 	    return ResourceKind.POD.getWeight();
 	}
+	
+	/**
+     * It will fetch pod parent by using of servicename and namespace, based on pod parent it will get  latestpodselector.
+     * Now it will start watching pod event changes on cluster based on namespace and latestpodselector.
+     * While watching pods it is reading status, here we are mainly reading three status (initialization, creations and readiness),
+     * after completion of each status we are showing status done message, whenever we get any pod has failed status immediately come out from watch
+     * and show deployment fail message. 
+     * 
+     * @param apiClient
+     * @param appName
+     * @param serviceName
+     * @param namespace
+     */
+
+	public void watch(ApiClient apiClient, String appName, String serviceName, String namespace)
+			throws HyscaleException {
+		 OkHttpClient httpClient=getUpdatedHttpClient(apiClient.getHttpClient());
+		 apiClient.setHttpClient(httpClient);
+		 CoreV1Api api = new CoreV1Api(apiClient);
+
+		String selector = ResourceSelectorUtil.getServiceSelector(appName, serviceName);
+		PodParent podParent=getPodParent(apiClient, serviceName, namespace);
+
+		Integer replicas = null;
+		String latestPodSelector = null;
+		if (podParent.getKind().equals(ResourceKind.STATEFUL_SET.getKind())) {
+			V1StatefulSetHandler v1StatefulSetHandler=(V1StatefulSetHandler) PodParentFactory.getHandler(podParent.getKind());
+			V1StatefulSet v1StatefulSet = (V1StatefulSet) podParent.getParent();
+			latestPodSelector = v1StatefulSetHandler.getPodSelector(null, v1StatefulSet, selector);
+			replicas = v1StatefulSetHandler.getReplicas(v1StatefulSet);
+		} else if (podParent.getKind().equals(ResourceKind.DEPLOYMENT.getKind())) {
+			V1DeploymentHandler v1DeploymentHandler=(V1DeploymentHandler) PodParentFactory.getHandler(podParent.getKind());
+			V1Deployment v1Deployment = (V1Deployment) podParent.getParent();
+			latestPodSelector = v1DeploymentHandler.getPodSelector(apiClient, v1Deployment, selector);
+			replicas = v1DeploymentHandler.getReplicas(v1Deployment);
+		}
```

Comment:
we can replace this by using handler from PodParentFactory.getHandler(podParent.getKind)